torch/distributed/tensor/_sharding_prop.py

```diff
@@ -335,7 +336,25 @@ def propagate_op_sharding_non_cached(self, op_schema: OpSchema) -> OutputShardin
         # special case op, we don't need to propagate for local
         # scalar. TODO: figure out a better way to handle this
         if op_schema.op is aten._local_scalar_dense.default:
-            return OutputSharding(None, op_schema)
+            # Need to Replicate() the input before calling .item() on the local tensor,
+            # to resolve any Partial(...) placements.
+            arg_spec = op_schema.args_spec[0]
+            placements = arg_spec.placements
+            return OutputSharding(
+                None,
+                OpSchema(
+                    op_schema.op,
+                    (
+                        DTensorSpec(
+                            arg_spec.mesh,
+                            tuple(Replicate() for _ in range(len(placements))),
+                            arg_spec.tensor_meta,
+                        ),
+                    ),
+                    {},
+                ),
+                needs_redistribute=True,
```

Comment:
nit: maybe only generate the op schema and set `needs_redistribute` when the placements is not all replicated